### PR TITLE
Obtains indices.

### LINKED ISSUE #84: Create base AsyncTask to obtain index values

### PR DESCRIPTION
**Pull request by DanMargera**
Task to obtain Selic, INPC and IPCA created.
Fixes several bugs in Indice Cursor Adapter and class Util.
Closes #84

### CODE REVIEW

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        setupIndiceCV(poupanca, IndiceEnum.POUPANCA);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
+
+        ContentValues IGPM = new ContentValues();
+        setupIndiceCV(IGPM, IndiceEnum.IGPM);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
```
Comment (tabocu) [inline app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: extract the numbers to const

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        setupIndiceCV(CDI, IndiceEnum.CDI);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
+
+        ContentValues poupanca = new ContentValues();
+        setupIndiceCV(poupanca, IndiceEnum.POUPANCA);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
```
Comment (tabocu) [inline app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: extract the numbers to const

Comment (tabocu) [reply app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: extract the numbers to const

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    protected List<ContentValues> doInBackground(Void... voids) {
+
+        String selicJsonStr = getJsonStr(SELIC_URL);
+        String inpcJsonStr = getJsonStr(INPC_URL);
+        String ipcaJsonStr = getJsonStr(IPCA_URL);
+
+        //placeholders
+
+        ContentValues CDI = new ContentValues();
+        setupIndiceCV(CDI, IndiceEnum.CDI);
```
Comment (tabocu) [inline app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: Make the function return the ContentValues like that:
`ContentValues CDI = setupIndiceCV(CDI, IndiceEnum.CDI);`

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        if (result.size() > 0) {
+            m_context.getContentResolver().delete
+                    (FinantialContract.IndiceEntry.CONTENT_URI, null, null);
+            ContentValues[] cvArray = result.toArray(new ContentValues[result.size()]);
+            m_context.getContentResolver().bulkInsert
+                    (FinantialContract.IndiceEntry.CONTENT_URI, cvArray);
+        }
+    }
+
+    private String getJsonStr(String URL) {
```
Comment (tabocu) [inline app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: Dont you think that this function could be extracted? It is the same for all json requests

**Review by tabocu — APPROVED** [review]
Since, those are simple changes, I will aprove it but, please, change it before merging

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java (hunk 2 above)
Comment (DanMargera) [reply app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: This is a placeholder, going to be removed as soon as the rest of the indices are implemented.

File: app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java (hunk 1 above)
Comment (DanMargera) [reply app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java]: This is a placeholder, going to be removed as soon as the rest of the indices are implemented.

**Pull request merged**